### DISCUSSION TITLE
Support for OpenTelemetry

### DISCUSSION THREAD

**Discussion by BartHuls** [akkadotnet/akka.net#7162] [Q&A] [answered]
I'm looking at moving to a OpenTelemetry implementation for my project.
Is there a change that Akka.net will also support this.
The current OpenTelemetry support does not support the actors we are using.
Maybe I'm doing something wrong, let me know.

Regards, Bart

**Comment by Aaronontheweb** [comment, marked as answer]
We have a commercial add-on that does this well at large scales: https://phobos.petabridge.com/

**Reply by BartHuls** [reply]
Hmm Phobos... no go... pricing is too much. 
OpenTelemetry is quit well supported by all Microsoft Libraries. which is mean and clean. 
Maybe it's cheaper to move to the MS Actor (Orleans) framework

**Reply by Aaronontheweb** [reply]
If $4000 / yr for accurate + high performance metrics and tracing complete with noise suppression and no need to do any instrumentation work is "too expensive," then that tells me you probably don't understand how hard this is to do yourself and how bad the default OpenTelemetry instrumentations are at all of the above in most OSS libraries.

**Reply by Aaronontheweb** [reply]
If you want to understand how Phobos works and why it's a cut above, see https://www.youtube.com/watch?v=ibqH1lRVXeU

And our benchmarks comparing uninstrumented Akka.NET vs. w/ Phobos:  https://phobos.petabridge.com/articles/performance.html

And our built-in dashboards: https://phobos.petabridge.com/articles/dashboards/prometheus-dashboard.html
